05/app.js

```diff
@@ -0,0 +1,40 @@
+function Student(initFirstName, initLastName) {
+    this.firstName = initFirstName;
+    this.lastName = initLastName;
+    this.grades = {};
+}
+
+Student.prototype.addGrade = function(subject, grade) {
+    if (typeof this.grades[subject] === 'undefined') {
+        this.grades[subject] = [];
+    }
+
+    this.grades[subject].push(grade);
+}
+
+Student.prototype.getAverageGrade = function(subject) {
+    if (typeof this.grades[subject] !== 'undefined') {
+
+        const avg = arr => arr.reduce((a, b) => a + b) / arr.length;
+        const getAvg = avg(this.grades[subject]);
+        console.log(this.grades)
+        console.log(`Average for ${this.grades.subject} is ${getAvg}`)
```

Comment:
Powinno być `this.grades[subject]` tutaj pobierana jest wartość dla zmiennej `subject`.
W Twoim przykładzie jest pobierana wartość dla właściwości o nazwie `subject`, a ona faktycznie jest równa `undefined`.